bindings/azure/cosmosdb/cosmosdb.go

```diff
@@ -85,30 +88,24 @@ func (c *CosmosDB) Init(metadata bindings.Metadata) error {
 		}
 	}
 
-	dbClient, err := client.NewDatabase(m.Database)
-	if err != nil {
-		return err
-	}
 	// Create a container client
-	dbContainer, err := dbClient.NewContainer(m.Collection)
+	dbContainer, err := client.NewContainer(m.Database, m.Collection)
 	if err != nil {
 		return err
 	}
 	if dbContainer.ID() == "" {
```

Comment:
nit: You can know this by checking the `m.Collection` and/or `m.Database` parameters, the `dbContainer.ID()` is just returning the value of `m.Collection`.

Reply:
NewContainer does not do any network calls (they don't have any `ctx` parameter), it just creates a client you can use to interact with a Container and in your case, call the `Read` method as it is done after this line.

My point is, before doing the Read call, this code is checking `dbContainer.ID() == ""`, basically validating if the `m.Collection` parameter is not `""`, it does not check that the container does not exist.

If the intent of this block is to check if the container does not exist, then it should be:

```
// Assuming m.Database and m.Collection will never be empty strings
// Otherwise, validate before calling NewContainer that m.Database or m.Collection
dbContainer, err := client.NewContainer(m.Database, m.Collection)
if err != nil {
	return err
}

c.client = dbContainer
ctx, cancel := context.WithTimeout(context.Background(), timeoutValue*time.Second)
containerRead, err := c.client.Read(ctx, nil)
cancel()
if err != nil {
    // If the container does not exist, it will return an err with Status 400
	return err
}

return nil
```

Reply:
In the past the error returned on `_, err := c.client.Read(ctx, nil)` was also the 429 error for too many metadata requests. Is this something that would now be handled internally in the SDK or could this also bubble up here?

If so, we do need add our backoff retry back to this and check for 429s specifically to reattempt the Read.

(FYI @ItalyPaleAle )

Reply:
@RyanLettieri  can you please fix this? At the very least, sounds like the check on this line is unnecessary because it doesn't actually do anything